uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -1227,4 +1228,169 @@ public function testDeleteAlsoDeletesLeaveRequestAndItsBalanceChangesFor() {
 
     $this->fail("Expected to not find the LeaveRequest with {$leaveRequest->id}, but it was found");
   }
+
+  public function testGetShouldOnlyReturnTheLeaveRequestsOfStaffMembersManagedByTheContactOnTheManagedByParam() {
+    $manager1 = ContactFabricator::fabricate();
+    $manager2 = ContactFabricator::fabricate();
+
+    $staffMember1 = ContactFabricator::fabricate();
+    $staffMember2 = ContactFabricator::fabricate();
+    $staffMember3 = ContactFabricator::fabricate();
+    $staffMember4 = ContactFabricator::fabricate();
+
+    // We need the contracts because LeaveRequest.get only returns Leave Requests
+    // overlapping contracts
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember1['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember2['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember3['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $staffMember4['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
```

Comment:
Contract fabrication could be easily refactored out into a single method, passing an array of staffmembers / staffmember id's and cycling through them, making the test easier to read.
```php
$this->createContracts([$staffmember1, $staffmember2, $staffmember3, $staffmember4]);
```
```php
private function createContracts($contacts) {
  foreach ($contacts as $staffMember) {
    HRJobContractFabricator::fabricate(
      ['contact_id' => $staffMember['id']],
      ['period_start_date' => '2016-01-01']
    );
  }
}
```